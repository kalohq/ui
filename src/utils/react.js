import ReactDOM from 'react-dom';

/**
 * Focus a ref’d react input when it is mounted
 * @param ref
 */
export function focusOnMount(ref) {
  const el = ReactDOM.findDOMNode(ref);

  if (el) {
    // Give focus and put the cursor at the end of
    // the value.
    el.focus();
    el.value = el.value;
  }
}

/**
 * Select the text of a ref’d react input when it is mounted
 * @param ref
 */
export function selectOnMount(ref) {
  if (ref && ref.select) {
    ref.select();
  }
}
