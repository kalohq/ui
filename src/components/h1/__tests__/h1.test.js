/* @flow */
/* eslint-env jest */
import * as React from 'react';
import {sheet} from 'emotion';
import serializer from 'jest-glamor-react';
import renderer from 'react-test-renderer';

import H1 from 'components/h1';

expect.addSnapshotSerializer(serializer(sheet));

describe('H1', () => {
  const defaultProps = {};
  const create = (props = {}) =>
    renderer
      .create(
        <H1 {...defaultProps} {...props}>
          An H1 Tag
        </H1>
      )
      .toJSON();

  test('should render shallow component ok', () => {
    const element = create();
    expect(element).toMatchSnapshot();
  });

  test('should allow weight to be overridden', () => {
    const element = create({
      weight: 'medium',
    });
    expect(element).toMatchSnapshot();
  });
});
