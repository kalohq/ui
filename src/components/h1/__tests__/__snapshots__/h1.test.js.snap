// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`H1 should allow weight to be overridden 1`] = `
<h1
  className="heading heading--extra-large heading--align-unset heading--ellipsis color-navy700 emotion-0 emotion-1"
  style={Object {}}
>
  An H1 Tag
</h1>
`;

exports[`H1 should render shallow component ok 1`] = `
<h1
  className="heading heading--extra-large heading--align-unset heading--ellipsis color-navy700 emotion-0 emotion-1"
  style={Object {}}
>
  An H1 Tag
</h1>
`;
